File: GO/Sully/Sully.go
package main
import("fmt";"os";"os/exec")
func main(){
i:=5;if(i==-1){return};if(os.Getenv("Sully")!=""){i--}
f:=fmt.Sprintf("Sully_%d.go",i);os.Setenv("Sully",f)
s:="package main%[1]cimport(%[2]cfmt%[2]c;%[2]cos%[2]c;%[2]cos/exec%[2]c)%[1]cfunc main(){%[1]ci:=%[4]d;if(i==-1){return};if(os.Getenv(%[2]cSully%[2]c)!=%[2]c%[2]c){i--}%[1]cf:=fmt.Sprintf(%[2]cSully_%%d.go%[2]c,i);os.Setenv(%[2]cSully%[2]c,f)%[1]cs:=%[2]c%[3]s%[2]c%[1]cfd,_:=os.Create(f);fmt.Fprintf(fd,s,0xa,0x22,s,i);fd.Close()%[1]cargs:=[]string{%[2]crun%[2]c,%[2]c./%[2]c+f};exec.Command(%[2]cgo%[2]c,args...).Start()%[1]c}%[1]c"
fd,_:=os.Create(f);fmt.Fprintf(fd,s,0xa,0x22,s,i);fd.Close()
args:=[]string{"run","./"+f};exec.Command("go",args...).Start()
}

File: C/Sully/Sully.c
#include<stdio.h>
#include<fcntl.h>
#include<unistd.h>
#include<stdlib.h>
int main() {
	int i=5;char f[32],cmd[128];
	if(i==0){return(0);}i-=((getenv("Sully"))!=0);
	char*s="#include<stdio.h>%1$c#include<fcntl.h>%1$c#include<unistd.h>%1$c#include<stdlib.h>%1$cint main() {%1$c%2$cint i=%6$d;char f[32],cmd[128];%1$c%2$cif(i==0){return(0);}i-=((getenv(%3$cSully%3$c))!=0);%1$c%2$cchar*s=%3$c%5$s%3$c;%1$c%2$csprintf(f,%3$cSully_%4$cd.c%3$c,i);setenv(%3$cSully%3$c,f,1);%1$c%2$cint fd=open(f,O_CREAT|O_RDWR,0744);%1$c%2$cdprintf(fd,s,0xa,0x9,0x22,0x25,s,i);%1$c%2$csprintf(cmd,%3$cgcc -Wall -Werror -Wextra %4$cs -o Sully_%4$cd && ./Sully_%4$cd%3$c,f,i,i);%1$c%2$cclose(fd);system(cmd);%1$c}%1$c";
	sprintf(f,"Sully_%d.c",i);setenv("Sully",f,1);
	int fd=open(f,O_CREAT|O_RDWR,0744);dprintf(fd,s,0xa,0x9,0x22,0x25,s,i);
	sprintf(cmd,"gcc -Wall -Werror -Wextra %s -o Sully_%d && ./Sully_%d",f,i,i);
	close(fd);system(cmd);
}
